frontend/src/components/Authentication.tsx

```diff
@@ -0,0 +1,24 @@
+import useAuth from "../hook/useAuth";
+import {Navigate, useLocation} from "react-router-dom";
+import React from "react";
+
+export default function Authentication ({children}:{children: React.ReactNode}) {
+
+    const location = useLocation();
+    const {user, isReady} = useAuth();
+
+    const navigate = (
+        <Navigate
+            to={isReady && !user ?
+                `/login?redirect=${encodeURIComponent(location.pathname + location.search)}`
+                : "/"}/>
+    );
+
+    if (isReady && user) {
+        return <>{children}</>
```

Comment:
Missing semi-colons at the end of the line!

Reply:
_Changed_